components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+        callbackUrl.setDisplayOrder(3);
+        configProperties.add(callbackUrl);
+
+        Property additionalQueryParams = new Property();
+        additionalQueryParams.setName(UAEPassAuthenticatorConstants.UAE.QUERY_PARAMS);
+        additionalQueryParams.setDisplayName("Additional Query Parameters");
+        additionalQueryParams.setRequired(false);
+        additionalQueryParams.setDescription("Add the additional query parameters");
+        additionalQueryParams.setType(UAEPassAuthenticatorConstants.UAEPassPropertyConstants.TEXTBOX);
+        additionalQueryParams.setDisplayOrder(4);
+        configProperties.add(additionalQueryParams);
+
+        Property logoutEnabled = new Property();
+        logoutEnabled.setName(UAEPassAuthenticatorConstants.UAE.LOGOUT_ENABLE);
+        logoutEnabled.setDisplayName("If Logout is enabled");
+        logoutEnabled.setRequired(false);
+        logoutEnabled.setDescription("Check here to enable the logout");
+        logoutEnabled.setType(UAEPassAuthenticatorConstants.UAEPassPropertyConstants.CHECKBOX);
+        logoutEnabled.setDisplayOrder(5);
+        configProperties.add(logoutEnabled);
+
+        Property isStagingEnv = new Property();
+        isStagingEnv.setName(UAEPassAuthenticatorConstants.UAE.UAEPASS_ENV);
+        isStagingEnv.setDisplayName("If Staging Environment");
+        isStagingEnv.setRequired(false);
+        isStagingEnv.setDescription("Check here, if using the Staging Environment.");
+        isStagingEnv.setType(UAEPassAuthenticatorConstants.UAEPassPropertyConstants.CHECKBOX);
+        isStagingEnv.setDisplayOrder(6);
+        configProperties.add(isStagingEnv);
+
+        return configProperties;
+    }
+
+
+    /**
+     * Redirects the user to the login page in order to authentication.
+     * In this UAE Pass Authenticator plugin, the user is redirected to the login page of the application which is
+     * configured in the UAE Pass side which acts as the external Identity Provider.
+     *
+     * @param request
+     * @param response
+     * @param context
+     * @throws AuthenticationFailedException - exception while creating the authorization code
+     */
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Authentication Request has initialized");
+            }
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            if (authenticatorProperties != null) {
+
+                String envUAEPass = getUAEPassEnvironment(context);
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("UAEPass " + envUAEPass + " Environment Selected");
```

Comment:
```suggestion
                    LOG.debug("UAEPass " + envUAEPass + " environment selected.");
```
Pay attention to case used in words. Check other places as well